core/src/main/java/com/linecorp/armeria/internal/common/util/ChannelUtil.java

```diff
@@ -310,37 +311,44 @@ public static InetSocketAddress localAddress(@Nullable Channel ch) {
             return null;
         }
 
-        final InetSocketAddress addr = findAddress(ch.localAddress());
-        if (addr != null) {
-            return addr;
-        }
-
         if (ch instanceof DomainSocketChannel) {
-            return findAddress(ch.remoteAddress());
+            return findAddress((DomainSocketChannel) ch);
+        } else {
+            return (InetSocketAddress) ch.localAddress();
         }
-
-        return null;
     }
 
     @Nullable
     public static InetSocketAddress remoteAddress(@Nullable Channel ch) {
         if (ch == null) {
             return null;
         }
-        return findAddress(ch.remoteAddress());
+
+        if (ch instanceof DomainSocketChannel) {
+            return findAddress((DomainSocketChannel) ch);
+        } else {
+            return (InetSocketAddress) ch.remoteAddress();
+        }
     }
 
     @Nullable
-    private static InetSocketAddress findAddress(@Nullable SocketAddress addr) {
-        if (addr instanceof InetSocketAddress) {
-            return (InetSocketAddress) addr;
+    private static DomainSocketAddress findAddress(DomainSocketChannel ch) {
+        final io.netty.channel.unix.DomainSocketAddress laddr = ch.localAddress();
```

Comment:
Question) What was the motivation for this change?

Just wanted to make sure, since if `ch.localAddress()` is always non-null, then `remoteAddress` may never be returned

Reply:
Not really. We don't return if `laddr.path()` returns null or an empty string. This method basically works around the peculiar behavior of Netty's `DomainSocketChannel`, which returns a `DomainSocketAddress` with an empty path.